Build failure: nano

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix-shell -p nano

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
openat-die.c:57:10: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
   56 |   error (exit_failure, errnum,
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   57 |          _("failed to return to initial working directory"));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
openat-die.c:31:18: note: expanded from macro '_'
   31 | #define _(msgid) dgettext ("gnulib", msgid)
      |                  ^
./gettext.h:112:39: note: expanded from macro 'dgettext'
  112 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
      |                                       ^
./error.h:496:39: note: expanded from macro 'error'
  496 |       __gl_error_call (error, status, __VA_ARGS__)
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./error.h:436:51: note: expanded from macro '__gl_error_call'
  436 |          __gl_error_call1 (function, __errstatus, __VA_ARGS__); \
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./error.h:421:26: note: expanded from macro '__gl_error_call1'
  421 |     ((function) (status, __VA_ARGS__), \
      |                          ^~~~~~~~~~~
openat-die.c:57:10: note: treat the string as an argument to avoid this
   57 |          _("failed to return to initial working directory"));
      |          ^
      |          "%s",
openat-die.c:31:18: note: expanded from macro '_'
   31 | #define _(msgid) dgettext ("gnulib", msgid)
      |                  ^
./gettext.h:112:39: note: expanded from macro 'dgettext'
  112 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
      |                                       ^
./error.h:496:39: note: expanded from macro 'error'
  496 |       __gl_error_call (error, status, __VA_ARGS__)
      |                                       ^
./error.h:436:51: note: expanded from macro '__gl_error_call'
  436 |          __gl_error_call1 (function, __errstatus, __VA_ARGS__); \
      |                                                   ^
./error.h:421:26: note: expanded from macro '__gl_error_call1'
  421 |     ((function) (status, __VA_ARGS__), \
      |                          ^
4 errors generated.
make[4]: *** [Makefile:4852: libgnu_a-openat-die.o] エラー 1
make[4]: *** 未完了のジョブを待っています....
make[4]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5/lib' から出ます
make[3]: *** [Makefile:6233: all-recursive] エラー 1
make[3]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5/lib' から出ます
make[2]: *** [Makefile:3093: all] エラー 2
make[2]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5/lib' から出ます
make[1]: *** [Makefile:2313: all-recursive] エラー 1
make[1]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5' から出ます
make: *** [Makefile:2254: all] エラー 2
error: builder for '/nix/store/5g3v2x5hp13bjai6hayyndnia1fcbppk-nano-8.5.drv' failed with exit code 2;
       last 25 log lines:
       >    31 | #define _(msgid) dgettext ("gnulib", msgid)
       >       |                  ^
       > ./gettext.h:112:39: note: expanded from macro 'dgettext'
       >   112 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
       >       |                                       ^
       > ./error.h:496:39: note: expanded from macro 'error'
       >   496 |       __gl_error_call (error, status, __VA_ARGS__)
       >       |                                       ^
       > ./error.h:436:51: note: expanded from macro '__gl_error_call'
       >   436 |          __gl_error_call1 (function, __errstatus, __VA_ARGS__); \
       >       |                                                   ^
       > ./error.h:421:26: note: expanded from macro '__gl_error_call1'
       >   421 |     ((function) (status, __VA_ARGS__), \
       >       |                          ^
       > 4 errors generated.
       > make[4]: *** [Makefile:4852: libgnu_a-openat-die.o] エラー 1
       > make[4]: *** 未完了のジョブを待っています....
       > make[4]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5/lib' から出ます
       > make[3]: *** [Makefile:6233: all-recursive] エラー 1
       > make[3]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5/lib' から出ます
       > make[2]: *** [Makefile:3093: all] エラー 2
       > make[2]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5/lib' から出ます
       > make[1]: *** [Makefile:2313: all-recursive] エラー 1
       > make[1]: ディレクトリ '/private/tmp/nix-build-nano-8.5.drv-0/nano-8.5' から出ます
       > make: *** [Makefile:2254: all] エラー 2
       For full logs, run:
         nix log /nix/store/5g3v2x5hp13bjai6hayyndnia1fcbppk-nano-8.5.drv
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/a4wf74piifb6cpmkyisrzjcac62dvf9q-source`

### Notify maintainers


@trofi 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/na/nano/package.nix
+++ b/pkgs/by-name/na/nano/package.nix
@@ -64,6 +64,10 @@ stdenv.mkDerivation rec {
         cp ${nixSyntaxHighlight}/nix.nanorc $out/share/nano/
       '';
 
+  # https://hydra.nixos.org/build/300187289/nixlog/1
+  # openat-die.c:57:10: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
+  hardeningDisable = [ "format" ];
+
   enableParallelBuilding = true;
   strictDeps = true;
 

